samples/WebApp/Code.gs

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * The GitHub OAuth2 client ID and secret. Uncomment and paste values from the
+ * GitHub developer settings.
+ */
+// var CLIENT_ID = '...';
+// var CLIENT_SECRET = '...';
+
+/**
+ * Handles GET requests. The page structure is described in the Page.html
+ * project file.
+ */
+function doGet() {
+  var service = getGitHubService();
+  var template = HtmlService.createTemplateFromFile('Page');
+  template.email = Session.getEffectiveUser().getEmail();
+  template.isSignedIn = service.hasAccess();
+  return template.evaluate()
+      .setTitle('Sample Web App')
+      .setSandboxMode(HtmlService.SandboxMode.IFRAME);
+}
+
+/**
+ * Builds and returns the authorization URL from the service object.
+ * @return {String} The authorization URL.
+ */
+function getAuthorizationUrl() {
+  return getGitHubService().getAuthorizationUrl();
+}
+
+/**
+ * Resets the API service, forcing re-authorization before
+ * additional authorization-required API calls can be made.
+ */
+function signOut() {
+  getGitHubService().reset();
+}
+
+/**
+ * Gets the user's GitHub profile.
+ */
+function getGitHubProfile() {
+  var service = getGitHubService();
+  if (!service.hasAccess()) {
+    throw new Error('Error: Missing GitHub authorization.');
+  }
+  var url = 'https://api.github.com/user';
+  var response = UrlFetchApp.fetch(url, {
+    headers: {
+      Authorization: 'Bearer ' + service.getAccessToken()
+    }
+  });
+  return JSON.parse(response.getContentText());
+}
+
+/**
+ * Gets the user's GitHub repos.
+ */
+function getGitHubRepos() {
```

Comment:
Combine `getGitHubRepos` and `getGitHubProfile`.
These method are too similar. I only see 1 difference (url).

Reply:
Kept them as separate methods, but combined the common bits.